File: Web/src/app/shared/entities/usersurvey.ts
export class UserSurvey {
  nickname: string;
  surveyId: number;
  userAnswers: UserAnswer[] = [];

  constructor(nickname: string, surveyId: number) {
    this.nickname = nickname;
    this.surveyId = surveyId;
  }
}

export class UserAnswer {
  questionId: number;
  answerId: number;

  constructor(questionId: number, answerId: number) {
    this.questionId = questionId;
    this.answerId = answerId;
  }
}

File: Api/Api/Data/Entities/UserSurvey.cs
namespace Api.Data.Entities
{
    public class UserSurvey
    {
        public int Id { get; set; }
        public string Nickname { get; set; }
        public int SurveyId { get; set; }
        public int QuestionId { get; set; }
        public int AnswerId { get; set; }
    }
}
